### PR TITLE
Softmax + error handling

### PR DESCRIPTION
**Pull request by pixelspark**
This PR combines #25 and #26 and implements proper error handling for softmax (in #25) as well. This might be a bit easier for you to merge.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -189,6 +199,10 @@ pub fn graph(
 
 pub fn model(graph: onnx::GraphProto) -> onnx::ModelProto {
     let mut model = crate::onnx::ModelProto::new();
+    let mut onnx_opset_import = OperatorSetIdProto::new();
+    onnx_opset_import.set_domain("".to_string());
+    onnx_opset_import.set_version(13);
```
Comment (haixuanTao) [inline src/utils.rs:204]: Great addition!

File: src/lib.rs [hunk 2]
```diff
@@ -155,7 +211,15 @@ impl Session {
...
             // Since contents are got in bytes, this converts these bytes back to f32
-            results.insert(output.clone(), bytemuck::cast_slice(&data).to_vec());
+
+            // The actual buffer may be bigger than what we should return, because buffers have a minimum size in wgpu
+            // Fetch the size we should expect so we can chop the buffer to the correct size
+            let output_buffer_size = len(&output.get_shape()) as usize;
+
+            results.insert(
+                output_name.to_string(),
+                bytemuck::cast_slice(&data)[..output_buffer_size].to_vec(),
```
Comment (haixuanTao) [inline src/lib.rs:221]: neat!

**Comment by haixuanTao** [comment]
Ok, so this is fine for me. 

I think the error handling is really going to help a lot! 

I think I made the mistake to merge the small binding index PR first and created some conflicts.

I believe we can merge after the conflict are resolved.

**Comment by pixelspark** [comment]
Thanks for the feedback! Merge conflicts are resolved, if CI finishes without error this is good to be merged 👍

**Comment by haixuanTao** [comment]
It is merged! Thanks for the massive Refactoring of error handling :100:

**Pull request merged**